src/components/Facility/FacilityUsers.tsx

```diff
@@ -12,44 +12,47 @@ import UserListView from "@/components/Users/UserListAndCard";
 
 import useFilters from "@/hooks/useFilters";
 
+import { RESULTS_PER_PAGE_LIMIT } from "@/common/constants";
+
 import routes from "@/Utils/request/api";
 import query from "@/Utils/request/query";
 
 export default function FacilityUsers(props: { facilityId: number }) {
   const { t } = useTranslation();
   const { qParams, updateQuery, Pagination } = useFilters({
-    limit: 18,
+    limit: RESULTS_PER_PAGE_LIMIT,
     cacheBlacklist: ["username"],
   });
   const [activeTab, setActiveTab] = useState(0);
   const { facilityId } = props;
 
   const { data: userListData, isLoading: userListLoading } = useQuery({
-    queryKey: ["facilityUsers", facilityId],
+    queryKey: ["facilityUsers", facilityId, qParams],
     queryFn: query(routes.facility.getUsers, {
       pathParams: { facility_id: facilityId },
+      queryParams: {
+        username: qParams.username,
+        limit: qParams.limit,
+        offset: (qParams.page - 1) * qParams.limit,
+      },
     }),
     enabled: !!facilityId,
   });
 
-  if (userListLoading) {
-    return (
-      <div className="px-6">
-        <Skeleton className="h-8 w-32 mb-4" />
-        <div className="flex items-center mb-4">
-          <Skeleton className="h-16 w-16 rounded-lg mr-3" />
-          <div>
-            <Skeleton className="h-4 w-14 mb-1" />
-            <Skeleton className="h-12 w-8" />
-          </div>
-        </div>
-        <div className="flex items-center justify-between mb-4">
-          <Skeleton className="h-10 w-72" />
-          <div className="flex space-x-2">
-            <Skeleton className="h-10 w-32" />
-            <Skeleton className="h-10 w-32" />
-          </div>
-        </div>
```

Comment:
Why did we remove skeleton loading?

Reply:
It's inside the return now @rithviknishad

Reply:
@Jeffrin2005 The point is to get the skeleton loader working within the return statement, not revert the changes 😅  Note how component behaves for search in encounters page vs in users.